sequencer/onchain.go

```diff
@@ -165,21 +165,52 @@ func (s *Sequencer) pushTransitionToContract(processID []byte,
 	); err != nil {
 		log.Debugw("failed to simulate state transition",
 			"error", err,
-			"pid", hex.EncodeToString(processID))
+			"pid", fmt.Sprintf("%x", processID))
 	}
 
 	// Submit the proof to the contract
-	txHash, err := s.contracts.SetProcessTransition(processID,
-		abiProof,
-		abiInputs,
-		blobSidecar,
-	)
-	if err != nil {
-		return fmt.Errorf("failed to submit state transition: %w", err)
+	log.Infow("state transition pending to be mined",
+		"pid", fmt.Sprintf("%x", processID))
+	if err := s.contracts.SetProcessTransitionAndWait(processID, abiProof,
+		abiInputs, blobSidecar, time.Hour*2, func(err error) {
```

Comment:
2 hours seems maybe too much?  Either way, we should declare all timeouts as constants or global variables part of the configuration. 

Furthermore, this Function is quite big, I'd be better if you can move this code to a separate function and then call it as callback with SetProcessTransitionAndWait()